### DISCUSSION TITLE
Number of points for curved structures

### DISCUSSION THREAD

**Discussion by manoranjanminz** [gdsfactory/gdsfactory#2872] [General]
Hi everyone,

I want to ask that when we put curved structures on a layout, for example a 90-degree circular bend or a microring, what should be the number of points of the polygon and on what factors does it depend on? 

It would be very helpful if there's any formula to determine it.

Regards,
Manoranjan

**Comment by joamatab** [comment]
we have a parameter PDK.bend_points_distance that you can customize

**Reply by manoranjanminz** [reply]
@joamatab I wanted to what is the ideal number of points for the curved shapes.

**Reply by tvt173** [reply]
it's dependent on multiple factors. for example
- what is your desired precision? this should never be finer than the database unit (DBU) of your layout, but you may want a coarser discretization to save on memory
- what kind of bend? for a plain bend with constant curvature, it is possible to derive a formula, but it is much more difficult for a bend with variable curvature, like an euler bend
- the radius (and potentially rate of change of curvature) of the curves

you can see that the attribute Joaquin pointed you to gets used to determine the number of points by default in a few bend functions like this `abs(int(angle / 360 * radius / PDK.bend_points_distance / 2))`. when extruding a path, you can also pass the `simplify` flag to reduce the number of points to a given precision. please be aware that this is a bit computationally expensive, but it may make sense if you want an optimal number of points on a complex curve (with variable curvature)
